### PR TITLE
feat: implement 0x3A - GASPRICE Opcode

### PR DESCRIPTION
**Pull request by TAdev0**
## Pull Request type

Feature

## What is the current behavior?

Resolves: #126 

## What is the new behavior?

GASPRICE opcode

## Does this introduce a breaking change?

 No

### CODE REVIEW

File: crates/evm/src/tests/test_instructions/test_environment_information.cairo [hunk 1]
```diff
@@ -41,3 +41,17 @@ fn test__exec_callvalue() {
     assert(ctx.stack.len() == 1, 'stack should have one element');
     assert(ctx.stack.pop().unwrap() == callvalue(), 'should be `123456789');
 }
+
+#[test]
+#[available_gas(20000000)]
+fn test_gaslimit() {
```
[the remark this answers is not in the document]
Comment (TAdev0) [reply crates/evm/src/tests/test_instructions/test_environment_information.cairo]: thank you @Eikix

File: crates/evm/src/tests/test_instructions/test_environment_information.cairo [hunk 2]
```diff
@@ -41,3 +41,17 @@ fn test__exec_callvalue() {
     assert(ctx.stack.len() == 1, 'stack should have one element');
     assert(ctx.stack.pop().unwrap() == callvalue(), 'should be `123456789');
 }
+
+#[test]
+#[available_gas(20000000)]
+fn test_gasPrice() {
```
Comment (enitrat) [inline crates/evm/src/tests/test_instructions/test_environment_information.cairo:47]:
```suggestion
fn test_gasprice() {
```

**Review by enitrat — APPROVED** [review]
lgtm, just change test name

**Review by enitrat — COMMENTED** [review]
actually @TAdev0 part of this work has already been implemented as part of #191, which required adding a gas_limit field to the ExecutionContext struct. You can rebase your work based on main, and just keep the opcode + the test

**Comment by TAdev0** [comment]
@enitrat  now it should be good :)

**Review by enitrat — APPROVED** [review]
lgtm

**Comment by enitrat** [comment]
@TAdev0 needs a rebase and then good to go

**Review by enitrat — APPROVED** [review]

**Pull request merged**